app/src/main/java/com/jerboa/ui/components/blocks/BlocksActivity.kt

```diff
@@ -0,0 +1,275 @@
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.lazy.rememberLazyListState
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.rounded.Close
+import androidx.compose.material3.ButtonDefaults
+import androidx.compose.material3.CircularProgressIndicator
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.Icon
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.SnackbarHostState
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextButton
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.runtime.setValue
+import androidx.compose.runtime.toMutableStateList
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.jerboa.R
+import com.jerboa.api.API
+import com.jerboa.api.ApiState
+import com.jerboa.api.toApiState
+import com.jerboa.model.SiteViewModel
+import com.jerboa.showBlockCommunityToast
+import com.jerboa.showBlockInstanceToast
+import com.jerboa.showBlockPersonToast
+import com.jerboa.ui.components.common.ApiEmptyText
+import com.jerboa.ui.components.common.ApiErrorText
+import com.jerboa.ui.components.common.CircularIcon
+import com.jerboa.ui.components.common.JerboaSnackbarHost
+import com.jerboa.ui.components.common.LoadingBar
+import com.jerboa.ui.components.common.SimpleTopAppBar
+import com.jerboa.ui.components.common.simpleVerticalScrollbar
+import com.jerboa.ui.theme.Title
+import it.vercruysse.lemmyapi.v0x19.datatypes.BlockCommunity
+import it.vercruysse.lemmyapi.v0x19.datatypes.BlockCommunityResponse
+import it.vercruysse.lemmyapi.v0x19.datatypes.BlockInstance
+import it.vercruysse.lemmyapi.v0x19.datatypes.BlockInstanceResponse
+import it.vercruysse.lemmyapi.v0x19.datatypes.BlockPerson
+import it.vercruysse.lemmyapi.v0x19.datatypes.BlockPersonResponse
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+
+private const val UNBLOCK_BUTTON_SIZE = 18
+
+@Composable
+private fun BlockedElementListItem(
+    apiState: ApiState<*>,
+    icon: String?,
+    name: String,
+    onUnblock: () -> Unit,
+    onSuccessfulUnblock: () -> Unit,
+) {
+    Row(verticalAlignment = Alignment.CenterVertically) {
+        icon?.let {
+            CircularIcon(
+                icon = it,
+                contentDescription = "",
+                size = 26.dp,
```

Comment:
I created a new size called SMALL_ICON_SIZE = 26.dp, but the naming doesn't seem intuitive since ICON_SIZE is equal to 20.dp. Do you think I should switch their names?

Reply:
Is it not possible to use the existing one of 28 dp?

We do all these presets for conformity. So that every page has the same colors, sizes, structure, ..

Reply:
@MV-GH I tried to search for a constant with 28.dp, couldn't find it

![image](https://github.com/LemmyNet/jerboa/assets/56604262/39512911-a86d-4203-bc96-de416f78c2b7)

Reply:
Ah mb, sure you can switch the naming